extensions/wikia/ThemeDesigner/ThemeDesignerHooks.class.php

```diff
@@ -43,6 +29,8 @@ public static function onUploadComplete( $image ) {
 			Wikia::invalidateFavicon();
 		}
 
+		self::resetThemeBackgroundSettings( $image->getTitle(), false );
```

Comment:
The previous hook handler would be called for every edit and would try to read title from master. But we only care about image uploads to the background image controlled by theme designer.